lib/internal/util/embedding.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+const { codes: { ERR_UNKNOWN_BUILTIN_MODULE } } = require('internal/errors');
+const { Module, wrapSafe } = require('internal/modules/cjs/loader');
+
+// This is roughly the same as:
+//
+// const mod = new Module(filename);
+// mod._compile(contents, filename);
+//
+// but the code has been duplicated because currently there is no way to set the
+// value of require.main to module.
+//
+// TODO(RaisinTen): Find a way to deduplicate this.
+
+function embedderRunCjs(contents) {
+  const filename = process.execPath;
+  const compiledWrapper = wrapSafe(filename, contents);
+
+  const customModule = new Module(filename, null);
+  customModule.filename = filename;
+  customModule.paths = Module._nodeModulePaths(customModule.path);
+
+  const customExports = customModule.exports;
+
+  embedderRequire.main = customModule;
+
+  const customFilename = customModule.filename;
+
+  const customDirname = customModule.path;
+
+  return compiledWrapper(
+    customExports,
+    embedderRequire,
+    customModule,
+    customFilename,
+    customDirname);
+}
+
+function embedderRequire(path) {
+  if (!Module.isBuiltin(path)) {
```

Comment:
I wonder if we want to enable `--expose-internals` for SEA or the embdder? (In that case the condition should be `BuiltinModule.canBeRequiredByUsers(id)`).

Reply:
I'm not sure about embedder because it might be a breaking change to disallow that.

However, since SEAs are new, I think it would be a good idea to not allow that from the beginning. This would prevent folks from using undocumented private APIs and running into problems. Instead if they really want to use something that can be accessed using `--expose-internals`, they can open an issue / PR and we can discuss if there's a better way of exposing the needed functionality ... but I'm not opposed to enabling the flag if there's a compelling use case.

Reply:
Yeah, I wouldn’t enable this unless we’re seeing a strong need for it.

Reply:
Not enabling this for now SGTM.